crates/bevy_pbr/src/render/view_transformations.wgsl

```diff
@@ -0,0 +1,198 @@
+#define_import_path bevy_pbr::view_transformations
+
+#import bevy_pbr::mesh_view_bindings as view_bindings
+
+/// World space:
+/// +y is up
+
+/// View space:
+/// -z is forward, +x is right, +y is up
+/// Forward is from the camera position into the scene.
+/// (0.0, 0.0, -1.0) is linear distance of 1.0 in front of the camera's view relative to the camera's rotation
+/// (0.0, 1.0, 0.0) is linear distance of 1.0 above the camera's view relative to the camera's rotation
+
+/// NDC (normalized device coordinate):
+/// https://www.w3.org/TR/webgpu/#coordinate-systems
+/// (-1.0, -1.0) in NDC is located at the bottom-left corner of NDC
+/// (1.0, 1.0) in NDC is located at the top-right corner of NDC
+/// Z is depth where: 
+///    1.0 is near clipping plane
+///    Perspective projection: 0.0 is inf far away
+///    Orthographic projection: 0.0 is far clipping plane
+
+/// UV space:
+/// 0.0, 0.0 is the top left
+/// 1.0, 1.0 is the bottom right
+
+
+// -----------------
+// TO WORLD --------
+// -----------------
+
+/// Convert a view space position to world space
+fn position_view_to_world(view_pos: vec3<f32>) -> vec3<f32> {
+    let world_pos = view_bindings::view.view * vec4(view_pos, 1.0);
+    return world_pos.xyz;
+}
+
+/// Convert a clip space position to world space
+fn position_clip_to_world(clip_pos: vec4<f32>) -> vec3<f32> {
+    let world_pos = view_bindings::view.inverse_view_proj * clip_pos;
+    return world_pos.xyz;
+}
+
+/// Convert a ndc space position to world space
+fn position_ndc_to_world(ndc_pos: vec3<f32>) -> vec3<f32> {
+    let world_pos = view_bindings::view.inverse_view_proj * vec4(ndc_pos, 1.0);
+    return world_pos.xyz / world_pos.w;
+}
+
+/// Convert a view space direction to world space
+fn direction_view_to_world(view_dir: vec3<f32>) -> vec3<f32> {
+    let world_dir = view_bindings::view.view * vec4(view_dir, 0.0);
+    return world_dir.xyz;
+}
+
+/// Convert a clip space direction to world space
+fn direction_clip_to_world(clip_dir: vec4<f32>) -> vec3<f32> {
+    let world_dir = view_bindings::view.inverse_view_proj * clip_dir;
+    return world_dir.xyz;
+}
+
+// -----------------
+// TO VIEW ---------
+// -----------------
+
+/// Convert a world space position to view space
+fn position_world_to_view(world_pos: vec3<f32>) -> vec3<f32> {
+    let view_pos = view_bindings::view.inverse_view * vec4(world_pos, 1.0);
+    return view_pos.xyz;
+}
+
+/// Convert a clip space position to view space
+fn position_clip_to_view(clip_pos: vec4<f32>) -> vec3<f32> {
+    let view_pos = view_bindings::view.inverse_projection * clip_pos;
+    return view_pos.xyz / view_pos.w;
+}
```

Comment:
Again, this one seems odd, except now it is applying a perspective division, whereas above it wasn't.

Reply:
If this weren't doing the perspective division, and so we consistently consider clip space to be ndc homogeneous pre-perspective divide (i.e. w is allowed to be something other than 1, though it may also be 1) then I think this is ok. Was the perspective divide a typo?

Reply:
woops! Yeah the `position_clip_to_world` was missing the perspective divide.  These should include it just like `position_ndc_to_***` does but where the input is a clip position.

Reply:
After looking into this further it was decided that the perspective divide isn't needed. (In most cases (all?) it is 1.0 so it wasn't doing anything anyway)